src/main/java/org/jenkinsci/plugins/credentialsbinding/impl/BindingStep.java

```diff
@@ -140,15 +143,16 @@
                 b.append(Pattern.quote(secret));
             }
             pattern = Secret.fromString(b.toString());
+            this.charsetName = charsetName;
         }
 
         @Override public OutputStream decorateLogger(AbstractBuild _ignore, final OutputStream logger) throws IOException, InterruptedException {
             final Pattern p = Pattern.compile(pattern.getPlainText());
             return new LineTransformationOutputStream() {
                 @Override protected void eol(byte[] b, int len) throws IOException {
-                    Matcher m = p.matcher(new String(b, 0, len));
+                    Matcher m = p.matcher(new String(b, 0, len, Charset.forName(charsetName)));
```

Comment:
Actually, I have anyway pushed what I think would fix the issue. When I am able to reproduce it I'd like to write a test for it...